### DISCUSSION TITLE
References from child models to parent models trigger system check errors?

### DISCUSSION THREAD

**Discussion by AstraLuma** [jazzband/django-polymorphic#694] [Q&A]
```python
class Parent(PolymorphicModel):
    ...

class Child(Parent):
    spam = models.ManyToManyField(Parent)
```

```
$ ./manage.py makemigrations
SystemCheckError: System check identified some issues:

ERRORS:
myapp.Child.spam: (fields.E305) Reverse query name for 'myapp.Child.spam' clashes with reverse query name for 'myapp.Child.parent_ptr'.
	HINT: Add or change a related_name argument to the definition for 'myapp.Child.spam' or 'myapp.Child.parent_ptr'.
```

**Comment by AdamDonna** [comment]
I'm pretty sure this isn't an issue that's specific to django-polymorphic https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.ForeignKey.related_query_name

Does this fix your issue? Could this issue be closed?

**Comment by AstraLuma** [comment]
I honestly don't remember what I was trying to do at the time. Does this kind of self-reference work now?

**Comment by bckohan** [comment]
@AstraLuma This is not really a problem with django-polymorphic. In Django multi-table model inheritance a pointer is stored as a OneToOne relation from children to parents. By default django uses the lowercased name of the child model for this relation from the parent to the child. When you add a ManyToMany to the child Django also uses the lowercase name of the child model as the reverse relation on the parent model. The error you are seeing is Django's system check system identifying the name clash. To fix it, you simply need to add a related_name to the spam field:

```python
class Parent(PolymorphicModel):
    field = models.CharField(max_length=20)

class Child(Parent):
    spam = models.ManyToManyField(Parent, related_name="spams")

# You would then be able to access related Child rows from the Parent like so:

Parent.objects.first().spams.all()
```
